akka-http-core/src/main/scala/akka/http/impl/engine/client/HttpsProxyGraphStage.scala

```diff
@@ -53,17 +55,15 @@ private final class HttpsProxyGraphStage(
   override def shape: BidiShape[ByteString, ByteString, ByteString, ByteString] = BidiShape.apply(sslIn, bytesOut, bytesIn, sslOut)
 
   private val connectMsg = {
-    val renderedProxyAuth = proxyAuthorization.map { httpCredentials =>
-      ByteString(s"${`Proxy-Authorization`(httpCredentials)}\r\n")
-    } getOrElse ByteString.empty
-
-    // format: OFF
-    ByteString(
-      s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\n" +
-      s"Host: $targetHostName\r\n") ++
-      renderedProxyAuth ++
-      ByteString("\r\n")
-    // format: ON
+    val r = new ByteStringRendering(256)
+
+    r ~~ "CONNECT " ~~ targetHostName ~~ ':' ~~ targetPort ~~ " HTTP/1.1" ~~ CrLf
+    r ~~ "Host: " ~~ targetHostName ~~ CrLf
+    proxyAuthorization.foreach { creds =>
+      r ~~ `Proxy-Authorization`(creds) ~~ CrLf
```

Comment:
So since `Proxy-Authorization`'s `renderValue` explicitly renders the credentials instead of relying on `toString` this works as desired - good thing we have a test to keep this verified :+1: - as long as we remember to not just change the test when we change the logic :smile: